src/AlgebraicGeometry/Schemes/SpecOpen/Rings/Methods.jl

```diff
@@ -18,6 +18,7 @@ function restrict(
     V::AbsSpec{<:Ring, <:MPolyQuoLocRing};
     check::Bool=true
   )
+  @check
   check && isempty(V) && return zero(OO(V))
```

Comment:
You are missing that the emptyness test and the appropriate return value are part of the mathematics and the algorithm itself.
Testing isempty for non-empty V is not cheap (Groebner basis!), unless the value has already been cached. But depending on the setting, in which one calls the function, it is often inevitable to do this test. There are however numerous settings in which it is known beforehand from the construction that V is non-empty -- without having called isempty on V before. So check=false is an optimization here, but not in the direction that you were expecting.

Reply:
For the final outcome does not make a difference: `isempty` being `true` implies `iszero(one(OO(V)) == true`. So it can not return anything but zero.

Reply:
OK, I was indeed missing this! Which to me suggest that a (brief!) comment explaining this would be helpful :-)